bin/oneoff/move_user_storage_ids_to_dashboard.rb

```diff
@@ -0,0 +1,69 @@
+require File.expand_path('../../../pegasus/src/env', __FILE__)
+require src_dir 'database'
+require 'optparse'
+
+# To run: ruby bin/oneoff/move_user_storage_ids_to_dashboard.rb
+# To revert: ruby bin/oneoff/move_user_storage_ids_to_dashboard.rb --revert
+
+# Parse options
+options = {
+  revert: false,
+}
+
+OptionParser.new do |opts|
+  opts.banner = <<~BANNER
+    Usage: #{File.basename(__FILE__)} [options]
+
+    This moves the user_storage_ids table to the Dashboard DB and renames it to user_project_storage_ids
+
+    Options:
+  BANNER
+
+  opts.on('--revert',
+    'Will revert the Database to its previous state (drop the view and move the table back to Pegasus)'
+  ) do |revert|
+    options[:revert] = revert
+  end
+
+  opts.on('-h', '--help', 'Prints this help message') do
+    puts opts
+    exit
+  end
+end.parse!
+puts "Options: #{options}"
+options.freeze
+
+$revert = options[:revert]
+
+def migrate_user_storage_ids_to_dashboard
+  pegasus_user_storage_ids = "#{CDO.pegasus_db_name}__user_storage_ids".to_sym
+  dashboard_user_project_storage_ids = "#{CDO.dashboard_db_name}__user_project_storage_ids".to_sym
+
+  puts "Renaming table pegasus.user_storage_ids -> dashboard.user_project_storage_ids"
+  # Rename the table to move it from the Pegasus schema to Dashboard
+  # Also change the name from user_storage_ids to user_project_storage_ids
+  DB.rename_table(pegasus_user_storage_ids, dashboard_user_project_storage_ids)
+
+  puts "Creating view pegasus.user_storage_ids"
```

Comment:
Yes. For hopefully less than 1 second, some read/write requests to the old table name will fail and we will  return a 5XX HTTP response.

Reply:
That's a good question. I wonder if that would be a reason to wrap it in a transaction, @sureshc, what do you think?

Reply:
Unfortunately, multiple MySQL 5.7 DDL statements can’t be wrapped in a transaction to succed or fail as a unit.

If we were swapping in a new table, the way `gh-ost` does, we could lock the old table and release the lock as we swap in the new table to ensure absoltely zero queries are executed on the old table name during the transition, but because we’re replacing the old table with a View, that’s not possible.